app/js/question_controller.js

```diff
@@ -109,31 +139,49 @@
         answers[i].created = Utils.time_since(new Date(created));
       }
 
+      add_thread_header(question);
       var html = nunjucks.render('thread.html', {
+        user: user,
         results: answers
       });
-      document.getElementById('comment-list').innerHTML = html;
+      var list = document.getElementById('comment-list');
+      list.insertAdjacentHTML('beforeend', html);
     });
   }
 
   var QuestionController = {
     init: function() {
-      var form = document.getElementById('question_form');
-      form.addEventListener('submit', submit_comment);
+      var question_view = location.search ? true : false;
+
+      // if this is a question view, not arrived at from clicking/tapping
+      // ask a question, hide the introduction section.
+      if (question_view) {
+        document.getElementById('thread-introduction').classList.add('hide');
+      }
+
+      // we will need the user details whether the user is posting a question
+      // or just viewing a question so, load the user during init
+      User.get_user().then(function(response) {
+        var form = document.getElementById('question_form');
+        form.addEventListener('submit', submit_comment);
 
-      // handle dialog close events
-      dialog_handler();
+        user = response;
+        question_thread = document.getElementById('question-thread');
```

Comment:
If we're storing this, then we can replace all the `document.getElementById('question-thread')` in this file.